### PR TITLE
import migrations and don't require diesel_cli for admins

### PR DESCRIPTION
**Pull request by fdb-hiroshima**
this is an attempt at integrating migrations to plm, with the goal to remove need for diesel_cli, and to allow arbitrary rust in migrations (can come handy for many reasons, like need to upgrade tantivy index version due to upstream changes, without need to take special steps for sysadmins)

- [x] backbone for the feature
- [x] integration to plm as new sub-command
- [x] update ci
- [x] make plume fail to launch if it detect migrations must be run

### CODE REVIEW

File: script/run_unit_test.sh [hunk 1]
```diff
@@ -4,6 +4,10 @@ for file in target/debug/*-*[^\.d]; do
 	if [[ -x "$file" ]]
 	then
 		filename=$(basename $file)
+		if [[ "plume_macro-c53c304b7b109570" =~ ^plume_macro ]]; then
```
Comment (BaptisteGelez) [inline script/run_unit_test.sh]: Isn't this always true? (maybe not, I'm really bad with Bash)

Comment (fdb-hiroshima) [reply script/run_unit_test.sh]: oops I tested with a value, and forgot to use the variable after :grimacing:

File: plume-macro/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+                .read_to_string(&mut down_sql)
+                .unwrap();
+            let name = path
+                .file_name()
+                .unwrap()
+                .to_str()
+                .unwrap()
+                .chars()
+                .filter(char::is_ascii_digit)
+                .take(14)
```
Comment (igalic) [inline plume-macro/src/lib.rs]: this seems like a strange thing to `take()`
what is this whole expression supposed to do?

Comment (fdb-hiroshima) [reply plume-macro/src/lib.rs]: diesel store whether migrations were run by storing a 14 digit code. For `2019-03-06-115158_blog_images`, it is translated to `20190306115158`. To do that I get an iterator over the folder name, filter only digits, and take the first 14 ones (if there are more, they are ignored)

File: plume-models/src/migrations.rs [hunk 3]
```diff
@@ -53,7 +53,7 @@ impl ImportedMigrations {
         let latest_migration = conn.latest_run_migration_version()?;
         let latest_id = if let Some(migration) = latest_migration {
             self.0
-                .binary_search_by_key(&migration.as_str(), |mig| mig.name)
+                .binary_search_by_key(&dbg!(migration.as_str()), |mig| dbg!(mig.name))
```
Comment (BaptisteGelez) [inline plume-models/src/migrations.rs]: Should these `dbg!` be here?

Comment (fdb-hiroshima) [reply plume-models/src/migrations.rs]: no :x

**Comment by fdb-hiroshima** [comment]
At some point the documentation should also be updated to reflect that change

**Comment by fdb-hiroshima** [comment]
@BaptisteGelez I think it's possible, I'll try to do it (diesel_cli does that from what I saw in it's code)

**Comment by igalic** [comment]
> I don't think it would be possible, but creating the PostgreSQL database if it doesn't exist would be perfect.

usually, create db is a more privileged operation than creating anything in that db…
and i sure hope people aren't giving our software that kind of permissions 😅

**Comment by fdb-hiroshima** [comment]
So, shall I try, or do I assume people have a "secure" configuration that don't allow us to do privileged things?

**Comment by BaptisteGelez** [comment]
Let's say people have a secure configuration. :smile:

**Pull request merged**